lib/ProductOpener/Food.pm

```diff
@@ -1255,27 +1259,51 @@ sub normalize_quantity($) {
 =head2 normalize_serving_size($)
 
 Returns the size in g or ml for the serving. Eg.:
-normalize_serving_size(1 barquette de 40g) returns 40
-normalize_serving_size(2.5kg)              returns 2500
+normalize_serving_size(1 barquette de 40g)->returns 40
+normalize_serving_size(2.5kg)->returns 2500
 
 =cut
 
 sub normalize_serving_size($) {
-
 	my $serving = shift;
 
-	my $q = 0;
-	my $u;
-
-	if ($serving =~ /((\d+)(\.|,)?(\d+)?)( )?($units)\b/i) {
-		$q = lc($1);
-		$u = $6;
+	# Regex captures any <number>( )?<unit-identifier> group, but leaves allowances for a preceding
+	# token to allow for patterns like "One bag (32g)", "1 small bottle (180ml)" etc
+	if ($serving =~ /^(.*[ \(])?(?<quantity>(\d+)(\.|,)?(\d+)?)( )?(?<unit>\w+)\b/i) {
+		my $q = $+{quantity};
+        my $u = normalize_unit($+{unit});
```

Comment:
Do we still need a normalize_unit() function if the translations are in %unit_conversion_map ?